File: livingdoc-fixture-api/src/main/java/org/livingdoc/fixture/api/converter/Language.java
package org.livingdoc.fixture.api.converter;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;


/**
 * Overrides the used language ({@link java.util.Locale}) for the annotated field or parameter.
 * A full list of {@link TypeConverter} considering this annotation can be found in the documentation.
 * The {@link #value()} has to be specified according to the <a href="https://tools.ietf.org/html/bcp47">BCP 47</a>
 * standard.
 * <p>
 * <b>Examples for this are:</b> {@code de}, {@code de-DE}, {@code en-US} etc.
 *
 * @see java.util.Locale#forLanguageTag(String)
 * @since 2.0
 */
@Retention(RetentionPolicy.RUNTIME)
@Target({ ElementType.FIELD, ElementType.PARAMETER })
public @interface Language {
    String value();
}

File: livingdoc-api/src/main/kotlin/org/livingdoc/api/conversion/Language.kt
package org.livingdoc.api.conversion

/**
 * Overrides the used language ([java.util.Locale]) for the annotated field or parameter.
 * A full list of [TypeConverter] considering this annotation can be found in the documentation.
 * The [value] has to be specified according to the [BCP 47](https://tools.ietf.org/html/bcp47)
 * standard.
 *
 * **Examples for this are:** `de`, `de-DE`, `en-US` etc.
 *
 * @see java.util.Locale.forLanguageTag
 * @since 2.0
 */
@Target(AnnotationTarget.FIELD, AnnotationTarget.VALUE_PARAMETER)
annotation class Language(val value: String)
